maps_dashboards/public/components/map_container/map_container.tsx

```diff
@@ -95,6 +102,34 @@ export const MapContainer = ({
     };
   }, [layers]);
 
+  // Handle map bounding box change, it should update the search if "request data around map extent" was enabled
+  useEffect(() => {
+    function renderLayers() {
+      layers.forEach((layer: MapLayerSpecification) => {
+        // We don't send search query if the layer doesn't have "request data around map extent" enabled
+        if (
+          layer.type === DASHBOARDS_MAPS_LAYER_TYPE.OPENSEARCH_MAP ||
+          !layer.source.useGeoBoundingBoxFilter
+        ) {
+          return;
+        }
+        handleDataLayerRender(layer, mapState, services, maplibreRef, undefined);
+      });
+    }
+
+    const debouncedRenderLayers = debounce(renderLayers, 200);
```

Comment:
Why 200? Can you add a constant with description?

Reply:
@VijayanB No particular reason, `200ms` sounds a reasonable debounce delay, what do you think?

Reply:
Sorry i don't have strong opinion for or against. i am okay with 200ms. Can you add comment why is this required? This will be great if we face any problem in delay.